humility-core/src/arch_rv.rs

```diff
@@ -0,0 +1,249 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use crate::arch::{instr_operands, Register, RegisterField};
+use anyhow::Result;
+use num_traits::ToPrimitive;
+use std::collections::BTreeMap;
+use strum_macros::EnumIter;
+
+#[allow(non_camel_case_types)]
+#[derive(
+    Copy,
+    Clone,
+    Debug,
+    Hash,
+    FromPrimitive,
+    ToPrimitive,
+    PartialEq,
+    Eq,
+    Ord,
+    PartialOrd,
+    EnumIter,
+)]
+
+///
+/// See table 3.3 in the riscv debug spec.
+/// (https://raw.githubusercontent.com/riscv/riscv-debug-spec/master/riscv-debug-stable.pdf#table.3.3)
+/// The pc is read through the DPC register
+///
+/// THOU MUST ORDER THE REGISTERS ACCORDING TO ADDRESS!
+///
+pub enum RVRegister {
```

Comment:
We should also have `mstatush` here, along with the decoding in `RVRegister::fields()`

May also be worthwhile to add `mscratch`? As far as I'm aware we only use it when context switching to temporarily save a register, so it's not super important.

`mtval` may also be a good one.